xbmc/interfaces/json-rpc/schema/types.json

```diff
@@ -463,7 +465,8 @@
       "properties": {
         "name": { "type": "string", "required": true },
         "role": { "type": "string", "required": true },
-        "artistid": { "type": "integer", "required": true }
+        "roleid": { "$ref": "Library.Id", "required": true },
```

Comment:
Not sure that is right. GetRoles, similar to GetGenres, gets all the possible role values and ids pairs. In comparison GetSong returns a structure that in effect extends the artists/artistid data with additional people that contribute to the recording, not just role pairs but artist name, id and role pair.